### PR TITLE
PLAT-436. Fail camus job on MessageDecodeException

### PR DESCRIPTION
**Pull request by abermanov**
Added parameter for message decode errors threshold. By default it's 0 - camus is swallowing decode errors as it was
Added tests

### CODE REVIEW

File: camus-etl-kafka/pom.xml [hunk 1]
```diff
@@ -43,6 +44,11 @@
 			<groupId>org.apache.hadoop</groupId>
 			<artifactId>hadoop-client</artifactId>
 		</dependency>
+        <dependency>
+            <groupId>commons-httpclient</groupId>
+            <artifactId>commons-httpclient</artifactId>
+            <version>3.1</version>
```
Comment (srosenberg) [inline camus-etl-kafka/pom.xml]: use a parameter please

File: camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/CamusJob.java [hunk 2]
```diff
@@ -336,7 +337,8 @@ public boolean accept(Path path) {
             in.close();
         }
         String prefix = job.getJobName() + ".outputs.";
-        
+        Configuration conf = job.getConfiguration();
```
Comment (srosenberg) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/CamusJob.java]: conf is never used

File: camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlRecordReader.java [hunk 3]
```diff
@@ -252,10 +251,14 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
                     CamusWrapper wrapper;
                     try {
                         wrapper = getWrappedRecord(key.getTopic(), bytes);
-                    } catch (Exception e) {
+                    } catch (IOException e) {
+                        exceptionCount++;
+                        long errorSchemaThresold = getSchemaErrorsThresold(context);
+                        if (errorSchemaThresold > 0 && errorSchemaThresold <= exceptionCount) {
```
Comment (srosenberg) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlRecordReader.java]: Default condition should be errorSchemaThreshold == 0.  Thus, the job will fail upon schema error.
If we want to relax that condition, we can set it to a positive value, e.g.  errorSchemaThreshold == 5 means that up to 5 errors are swallowed; 6th error fails the job.

**Comment by srosenberg** [comment]
Is it possible to fix the indentation? It's really difficult to read the diffs.

**Comment by abermanov** [comment]
@srosenberg I'm sorry for the indentation. I always use spaces for indentation, but tabs are used in original code. It's looks good in my IDEA, but not the same in a browser. I'll use tabs here

File: camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlRecordReader.java (hunk 3 above)
Comment (abermanov) [inline camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlRecordReader.java]: OK! will be done

**Comment by abermanov** [comment]
Please, review new pull request https://github.com/placeiq/camus/pull/2 (all commits and new fixes are merged into 1 commit)

**Pull request closed without merging**